lua/haskell-snippets/table-util.lua

```diff
@@ -0,0 +1,35 @@
+---@mod haskell-snippets.table-util
+
+---@brief [[
+--- Extra utilities related to Lua tables.
+---@brief ]]
+
+local M = {}
+
+---Concatenates tables.
+---
+---@return table
+function M.concat(...)
+  local t = {}
+  for _, v in ipairs { ... } do
+    for _, v2 in ipairs(v) do
+      table.insert(t, v2)
+    end
+  end
+  return t
+end
+
+---Maps a function over a table.
+---
+---@param f fun(any):any
+---@param t table
+---@return table
+function M.map(f, t)
```

Comment:
This can be replaced with [`vim.tbl_map`](https://neovim.io/doc/user/lua.html#vim.tbl_map()).

Reply:
Done